src/ngamsPlugIns/ngamsPlugIns/eso/printhead.py

```diff
@@ -938,8 +928,8 @@ def getKeyDict(self,key,desc=0,inst=0):
         keyDict = HeadDict()
         node = self['nodes'].copy()
         for hk in hkeys:
-            if node.has_key(hk) and type(node[hk]) == type({}):
-                keys = node[hk].keys()
+            if hk in node and type(node[hk]) == type({}):
+                keys = list(node[hk].keys())
```

Comment:
Otherwise the comparison in line 935 below will always be `False`

```suggestion
                keys = tuple(node[hk].keys())
```